src/merkle/smt/full/concurrent/mod.rs

```diff
@@ -107,17 +107,26 @@ impl Smt {
 
             // Aggregate all node mutations
             node_mutations.extend(mutations_per_subtree.into_iter().flatten());
-
-            debug_assert!(!subtree_leaves.is_empty());
         }
 
-        // Finalize the mutation set with updated roots and mutations
-        MutationSet {
+        // Create mutation set - if no mutations occurred, all fields should indicate no changes
+        let mutation_set = MutationSet {
             old_root: self.root(),
-            new_root: subtree_leaves[0][0].hash,
+            new_root: if node_mutations.is_empty() {
+                self.root()
+            } else {
+                subtree_leaves[0][0].hash
+            },
```

Comment:
Optional nit: I would maybe define a separate variable above for the new root and then use it here - e.g.:
```rust
let new_root = if node_mutations.is_empty() {
    self.root()
} else {
    subtree_leaves[0][0].hash
};
```